workflows/multiomics/full_pipeline/main.nf

```diff
@@ -195,4 +159,48 @@ workflow test_wf {
         assert output_list.size() == 1 : "output channel should contain one event"
         assert output_list[0][0] == "merged" : "Output ID should be 'merged'"
       }
+  input_ch = viashChannel(testParams, config)
+    // store output value in 3rd slot for later use
+    // and transform for concat component
+    | map { tup ->
+      data = tup[1]
+      new_data = [ id: data.id, input: data.input ]
+      [tup[0], new_data, data] + tup.drop(2)
+    }
+
+    input_ch.branch{
+        human: it[0] == "human"
+        mouse: it[0] == "mouse"
+      }.set{ test2_input_channels }
+
+    human_ch = test2_input_channels.human 
+      | remove_modality.run(
+        args: [ modality: "atac" ]
+      )
+
+    mouse_ch = test2_input_channels.mouse |
+      remove_modality.run(
+        args: [ modality: "rna" ]
+      )
```

Comment:
I would replace with:

```groovy
    human_ch = input_ch
      | filter{it[0] == "human"}
      | remove_modality.run(
        args: [ modality: "atac" ]
      )

    mouse_ch = input_ch
      | filter{it[0] == "mouse"}
      | remove_modality.run(
        args: [ modality: "rna" ]
      )
```

`branch{}` and `set{}` are part of Nextflow DSL1, which I try to avoid.

Reply:
Good to know!